File: src/Services/Qualifying/Qualifying.Service.Queries/DTOs/StoredProcedure/GetTeacherDetail.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Qualifying.Service.Queries.DTOs.StoredProcedure
{
    public class GetTeacherDetailDto
    {
        public string? QuestionDescription { get; set; }
        public decimal QuestionScoreAverage { get; set; }
        public decimal QualificationAverage { get; set; }
        public int QualificationQuantity { get; set; }
        public string? CourseName { get; set; }
        public string? TeacherCodeName { get; set; }
        public string? TeacherPhoto { get; set; }
    }
}

File: src/app/infrastructure/dto/qualifying/get-teacher-detail.ts
import { TeacherDetail } from "src/app/domain/models/qualifying/teacher-detail";

export class GetTeacherDetailDto {
    questionDescription!: string;
    questionScoreAverage!: number;
    qualificationAverage!: number;
    qualificationQuantity!: number;
    courseName!: string;
    teacherCodeName!: string;
    teacherPhoto!: string;
}

export class GetTeacherDetailDtoTransformer {
    transform(dto: GetTeacherDetailDto[]): TeacherDetail[] {
        return dto.map(item => {
            let obj = new TeacherDetail();
            obj.questionDescription = item.questionDescription;
            obj.questionScoreAverage = item.questionScoreAverage;
            obj.qualificationAverage = item.qualificationAverage;
            obj.qualificationQuantity = item.qualificationQuantity;
            obj.courseName = item.courseName;
            obj.teacherCodeName = item.teacherCodeName;
            obj.teacherPhoto = item.teacherPhoto;
            return obj;
        });
    }
}
